department-of-veterans-affairs/caseflow
# Batch calls to users table

### Description
Removes cacophony of calls to find individual users when calling `find_all_attorneys` or `find_all_judges`

### Testing Plan
1. Checkout master
2. Run benchmark for old method and view said cacophony
```ruby
> require "benchmark"
> Benchmark.measure { AttorneyRepository.find_all_attorneys }
=> #<Benchmark::Tms:0x00007fc2abc1b7c0
 @cstime=0.0,
 @cutime=0.0,
 @label="",
 @real=2.404171000001952,
 @stime=0.13165599999999955,
 @total=1.6080920000000063,
 @utime=1.4764360000000067>
```
3. Checkout hschallhorn/speed-up-user-repos
4. Run benchmark for new method
```ruby
> reload!
> Benchmark.measure { AttorneyRepository.find_all_attorneys }
=> #<Benchmark::Tms:0x00007fc2b4d21aa8
 @cstime=0.0,
 @cutime=0.0,
 @label="",
 @real=0.2092659999616444,
 @stime=0.030889999999999418,
 @total=0.1086920000000049,
 @utime=0.07780200000000548>
```
5. Run all flake tests in dev to ensure they pass
